plugins/provisioners/ansible/cap/guest/alpine/ansible_install.rb

```diff
@@ -0,0 +1,45 @@
+require_relative "../facts"
+require_relative "../pip/pip"
+
+module VagrantPlugins
+  module Ansible
+    module Cap
+      module Guest
+        module Alpine
+          module AnsibleInstall
+
+            def self.ansible_install(machine, install_mode, ansible_version, pip_args, pip_install_cmd = "")
+              python_setup machine
+              case install_mode
+                when :pip
+                  pip_setup machine
+                  Pip::pip_install machine, "ansible", ansible_version, pip_args, true
+                when :pip_args_only
+                  pip_setup machine
+                  Pip::pip_install machine, "", "", pip_args, false
+                else
+                  ansible_apk_install machine
+              end
+            end
+
+            private
+
+            def self.python_setup(machine)
+              machine.communicate.sudo "apk add --update --no-cache python3"
+              machine.communicate.sudo "if [ ! -e /usr/bin/python ]; then ln -sf python3 /usr/bin/python ; fi"
+            end
+
+            def self.ansible_apk_install(machine)
+              machine.communicate.sudo "apk add --update --no-cache ansible"
+            end
+
+            def self.pip_setup(machine)
+              machine.communicate.sudo "pip3 install --upgrade pip"
```

Comment:
It looks like we do this for debian (and I also had to do this manually while testing the PR with the guest I was using), we'll want to install some required dev libraries for building ansible with pip:

```ruby
machine.communicate.sudo "apk add build-base libffi-dev python3-dev openssl-dev"
```

Reply:
hm, I successfully tested ansible installation via pip without need to install those dev libraries... Interesting...

Reply:
so we install those dev libs to install pip via get_pip.py script. In python3 we get pip3 with python3 package. In most cases, this command will run once, on first provision. 
so, yeah, I'll add those back since we have an option to run provision against already provisioned VM and it could fail...

Reply:
sorry, had a tough week :)
So, question still the same. Do we need pip_install_cmd? and do we really need those dev packages? Both required **ONLY** when we use `get_pip.py` script to install `pip`. For Alpine we don't use it and those parameters and packages not required since pip installed via package manager so it's already "compiled" and will install those deps only if required.

If you mean those libraries for later use with ansible then it must be handled by ansible role, not inside this provisioner.

Reply:
Hey there, no worries at all. Tough week for many of us, no need to apologize :heart: 

> So, question still the same. Do we need pip_install_cmd? and do we really need those dev packages?

Yep, please add this to your PR. If you want to see what other guest capabilities are doing, they are also adding this command. It makes the pip install more flexible in case users need to modify what pip command is there. We can do what debian and the other guest capabilities to and default it to empty string.

https://github.com/hashicorp/vagrant/blob/8c11489ea73ce2672d9c81a87dd2887eac99f5cd/plugins/provisioners/ansible/cap/guest/debian/ansible_install.rb#L38-L42

For the dev packages, I recommend installing them too. Especially since the most popular alpine boxes on vagrant cloud are missing them and end up failing when trying the pip install method with this PR. So I imagine the first thing users will run into are these missing packages. We also install them with Debian, so we should be good installing them here too.

Hope that makes some more sense.

Reply:
Again, your example uses `Pip::get_pip`, because debian's default version is 2.7 and it's required to install pip separately. And only this command consumes `pip_install_cmd`. 
My version does not need `Pip::get_pip` because `pip` installed as a part of python 3. And I don't need `pip_install_cmd` at all. That's why I removed that argument from pip_setup but
 left that in ansible_install to make it backward compatible.